feature_engine/sanity_check/similar_columns.py

```diff
@@ -0,0 +1,68 @@
+from typing import Any, List
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class SimilarColumns(BaseEstimator, TransformerMixin):
+    """Ensure that similar columns are in test and train dataset.
+
+
+    Parameters
+    ----
+    None
+
+    """
+
+    def __init__(self, impute_with: Any = np.nan):
+        self.impute_with = impute_with
+        self.col: List[str] = []
+
+    def fit(self, df: pd.DataFrame, y: pd.Series = None, **fit_params):
+        """Fit columns schema
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+        """
+
+        self.col = df.columns
```

Comment:
all parameters learned in fit, should have a trailing underscore as per sklearn convetion. Also note that we use variables_ and not cols in feature engine